File: 1492-the-kth-factor-of-n/1492-the-kth-factor-of-n.cpp
class Solution {
public:
    int kthFactor(int n, int k) {
        vector<int> factors;
        for(int i=1; i<=n; i++){
            if (n%i==0) factors.push_back(i);
            if (factors.size() == k) break;
        }
        if (k<= factors.size())
            return factors[k-1];
        else return -1;
    }
};

File: 1492-the-kth-factor-of-n/1492-the-kth-factor-of-n.py
class Solution:
    def kthFactor(self, n: int, k: int) -> int:
        l=[]
        for i in range(1,n+1):
            if n%i==0:
                l.append(i)
        if k<=len(l):
            return l[k-1]
        return -1